Clockwise.Redis/CircuitBreakerBrokerPartition.cs

```
@@ -111,9 +111,12 @@ private static CircuitBreakerStateDescriptor TryDeserialise(string serialised)
 
         public async Task Initialize(ISubscriber subscriber)
         {
-            keySpaceObserver = new KeySpaceObserver(dbId, key, subscriber);
-            keySpaceSubscription = keySpaceObserver.Subscribe(KeySpaceNotificationHandler);
-            await keySpaceObserver.Initialize();
+            if (keySpaceObserver == null)
```

This isn't concurrency safe.

---

Also, the no-op on repeat initialization is misleading. The signature seems to indicate I could initialize more than one subscriber, but after the first, I would get a no-op. Should we support more than one subscriber?

---

the partition uses a subscriber which is a Redis subscriber, this is invoked with one subscriber only soi this initialisation is one-off, also this class is internal only, not supposed to be used outside

---

That still doesn't prevent someone working on this codebase from thinking it can be subscribed more than once. Maybe throw an exception if it's already been initialized?